src/libappimage/core/impl/StreambufType2.cpp

```diff
@@ -0,0 +1,38 @@
+extern "C" {
+// system
+#include <sys/stat.h>
+
+// libraries
+#include <nonstd.h>
+#include <squashfuse.h>
+#include <squashfs_fs.h>
+}
+
+
+#include "appimage/core/Exceptions.h"
+#include "StreambufType2.h"
+
+using namespace appimage::core::impl;
+
+StreambufType2::StreambufType2(sqfs fs, const sqfs_inode& inode, unsigned long size)
```

Comment:
Yes. `squashfuse` functions don't work on a const "sqfs". A reference could be used but it will end copied to the property anyway.

Reply:
Can you put that in a comment? It's the kind of special knowlege only you have right now, as you implemented this, if I ever had to work on this I might accidentally break it because I am not aware of this.

Right now `fs` is copied twice, you may want to `fs(std::move(fs))` in your initializer list.

Reply:
Using a reference is not possible but a pointer can save us the extra copy.
Fixed